core/factories.py

```diff
@@ -96,3 +98,40 @@ def created_by(self):
     @factory.lazy_attribute
     def created_by_structure(self):
         return Structure.objects.get(libelle="Structure Test")
+
+
+class MessageFactory(DjangoModelFactory):
+    class Meta:
+        model = Message
+
+    message_type = FuzzyChoice([choice[0] for choice in Message.MESSAGE_TYPE_CHOICES])
+    title = factory.Faker("sentence", nb_words=10)
+    content = factory.Faker("paragraph")
+
+    sender = factory.SubFactory(ContactAgentFactory)
+
+    @factory.post_generation
+    def recipients(self, create, extracted, **kwargs):
+        if not create:
+            return
+        if extracted:
+            self.recipients.set(extracted)
+        else:
+            for i in range(0, random.randint(1, 10)):
+                if random.randint(1, 2) == 1:
+                    self.recipients.add(ContactAgentFactory())
+                else:
+                    self.recipients.add(ContactStructureFactory())
```

Comment:
petite suggestion d'amélioration du code :  
- suppression du else avant boucle for, 
- suppression du 0 dans le range,
- suppression du if et else dans la boucle et utilisation de `random.choice`
```suggestion
        if extracted:
            self.recipients.set(extracted)
            return
        for _ in range(random.randint(1, 10)):
        factory_class = random.choice([ContactAgentFactory, ContactStructureFactory])
        self.recipients.add(factory_class())
```